lib/datadog/core/configuration/components.rb

```diff
@@ -149,6 +151,18 @@ def shutdown!(replacement = nil)
           telemetry.stop!
           telemetry.emit_closing! unless replacement
         end
+
+        private
+
+        def register_remote_functionality
+          return unless remote
+
+          if appsec
+            Datadog::Core::Remote.register_capabilities(Datadog::AppSec::Remote.capabilities)
+            Datadog::Core::Remote.register_receivers(Datadog::AppSec::Remote.receivers)
+            Datadog::Core::Remote.register_products(Datadog::AppSec::Remote.products)
+          end
+        end
```

Comment:
This should be able to move to the `Client::Capabilities` instance I mentioned below, and test against the `settings` instance that gets passed to its initializer.

Rationale: we very much want to avoid static globals (and pseudo-globals such as class variables) as much as possible.